crawl-ref/source/l-spells.cc

```diff
@@ -97,6 +98,65 @@ LUAFN(l_spells_min_range)
     PLUARET(number, spell_range(spell, 0));
 }
 
+
+/*** If this spell is aimed at (x,y), what path will it actually take?
+ * @tparam string spell name
+ * @tparam int x coordinate to aim at, in player coordinates
+ * @tparam int y coordinate to aim at, in player coordinates
+ * @treturn string|nil a list in "x1,y1 x2,y2 x3,y3..." format giving the path
+ *    the spell will take, in player coordinates
+ * @function path
+ */
+LUAFN(l_spells_path)
+{
+    spell_type spell = spell_by_name(luaL_checkstring(ls, 1), false);
+    if (!you.has_spell(spell))
+    {
+        lua_pushnil(ls);
+        return 1;
+    }
+    int power = calc_spell_power(spell, true);
+    int range = spell_range(spell, power);
+    zap_type zap = spell_to_zap(spell);
+    if( power <= 0 || range <= 0 )
+    {
```

Comment:
A note on Crawl code style: `if` statements should be written like this: `if (condition_1 || condition_2)` with a space before the parentheses and no space between the parentheses and the conditions.